src/reconcile/rec_visibility.c

```diff
@@ -469,7 +463,18 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
      * Additionally history store reconciliation is not set skip saving an update.
      */
     if (__rec_need_save_upd(session, r, upd_select, has_newer_updates)) {
-        WT_ERR(__rec_update_save(session, r, ins, ripcip, upd_select->upd, upd_memsize));
+        /*
+         * We should restore the update chains to the new disk image if there are newer updates in
+         * eviction, or for cases that don't support history store, such as in-memory database and
+         * fixed length column store.
+         */
+        supd_restore = F_ISSET(r, WT_REC_EVICT) &&
+          (has_newer_updates || F_ISSET(S2C(session), WT_CONN_IN_MEMORY) ||
+                         page->type == WT_PAGE_COL_FIX);
+        if (supd_restore)
+            r->cache_write_restore = true;
+        WT_ERR(
+          __rec_update_save(session, r, ins, ripcip, upd_select->upd, supd_restore, upd_memsize));
```

Comment:
`upd_select.upd` is different to `supd_restore`. If a long update chain has uncommitted changes, the older updates are moved to HS and data file, the update chain needs to be restored because it has uncommitted updates. So the answer is yes.  There are SUPD entries that are both put into the HS and are restored

Reply:
In durable history, we never choose an onpage value which is not a standard update or a modify. I'll modify the code to assert that instead of test that.

Reply:
Looks like we do set upd_select.upd to tombstone and set onpage_upd to null

Reply:
I feel `upd_select.upd` and `upd_select.onpage_value`are mostly overlapping. I think it is possible to restructure the code to get rid of one of them. However, that should be done in a separate ticket.